src/Text/Layout/Table/LineStyle.hs

```diff
@@ -0,0 +1,167 @@
+module Text.Layout.Table.LineStyle where
+
+import Data.Function (on)
+
+-- | The line styles supported by the Unicode Box-Drawing block.
+data LineStyle
+    = NoLine
+    | SingleLine
+    | HeavyLine
+    | DoubleLine
+    | DashLine
+    | HeavyDashLine
+    | Dash4Line
+    | HeavyDash4Line
+    | Dash2Line
+    | HeavyDash2Line
+
+-- | Join styles supported by the Unicode Box-Drawing block.
+data UnicodeJoin = NoJoin | Light | Heavy | Double
```

Comment:
Is this not actually the Unicode line style?  Sounds confusing to me.

Reply:
The `UnicodeJoin` type is solely an internal thing, and is there as a work-saving measure. Unicode itself allows nine different line styles, but only includes joins for three of them. `UnicodeJoin` is a projection from `LineStyle` allowing us to handle all the joins Unicode supports, without having to handle the unsupported cases manually. As it is, handling all the joins requires about 135 cases: if we didn't introduce `UnicodeJoin` it would be more like 10 000 (order of magnitude).